src/business-logic/services/device-connection-service-implementation.ts

```diff
@@ -0,0 +1,106 @@
+import { DeviceConnectionService } from './interfaces/device-connection-service'
+import { Device } from '../../model/entities/device'
+import { DeviceConnectionFactory } from './interfaces/device-connection-factory'
+import { DeviceConnection } from './interfaces/device-connection'
+import { DeviceConnectionStatus } from '../../model/enums/device-connection-status'
+import { DeviceAlreadyConnectedException } from '../../model/exceptions/device-already-connected-exception'
+import { DeviceNotFoundException } from '../../model/exceptions/device-not-found-exception'
+import { Logger } from '@tv2media/logger/*'
+import { DeviceService } from './interfaces/device-service'
+
+type DeviceAggregate = {
+  deviceConnection: DeviceConnection,
+  device: Device
+}
+
+export class DeviceConnectionServiceImplementation implements DeviceConnectionService {
+  private readonly connectedDevices: Map<string, DeviceAggregate> = new Map()
+
+  constructor(private readonly deviceConnectionFactory: DeviceConnectionFactory, private readonly deviceService: DeviceService, private readonly logger: Logger) {
```

Comment:
We have the convention that when a logger is used in a context, the context tells the logger who is using the logger. This is done with `logger.tag(this.constructor.name)` which creates a new logging context.
By tagging the log entries with the class that produces them, it makes it easier to see where a log originated.
The way we usually do so is by not binding the passed in logger instance via the parameters, but instead set it in the constructor body, like so:
```typescript
export class DeviceConnectionServiceImplementation implements DeviceConnectionService {
  private readonly logger: Logger

  constructor(/* other depenencies */, logger: Logger) {
    this.logger = logger.tag(this.constructor.name)
  }
}
```
This also means that the debug statements in the constructor should use `this.logger` instead of `logger`.

Reply:
Changed to follow the procedure everywhere else.